metadatastore/commands.py

```diff
@@ -638,6 +638,29 @@ def find_events(**kwargs):
 
 
 @_ensure_connection
+def count_events(**kwargs):
+    """Like find_events, but it returns the count, not the actual Events.
+
+    This function is likely to be deprecated in this near future. Instead,
+    find_events will return an iterator that has a length.
+    """
+    # Some user-friendly error messages for an easy mistake to make
+    if 'event_descriptor' in kwargs:
+        raise ValueError("Use 'descriptor', not 'event_descriptor'.")
+    if 'event_descriptor_id' in kwargs:
+        raise ValueError("Use 'descriptor_id', not 'event_descriptor_id'.")
+
+    _format_time(kwargs)
+    try:
+        kwargs['descriptor_id'] = kwargs.pop('descriptor').id
+    except KeyError:
+        pass
+    _normalize_object_id(kwargs, '_id')
+    _normalize_object_id(kwargs, 'descriptor_id')
+    return len(Event.objects(__raw__=kwargs))
```

Comment:
Doing this on the HXN scan that used to take a long time is pretty snappy already. (Note that this is done with hash 4254230070efd1c897bc0763dae4895aa34068ad). 

```python
In [1]: from metadatastore.commands import *
/home/edill/miniconda/envs/ophyd/lib/python2.7/site-packages/pytz/__init__.py:29: UserWarning: Module bson was already imported from /home/edill/miniconda/envs/ophyd/lib/python2.7/site-packages/bson/__init__.pyc, but /home/edill/miniconda/envs/ophyd/lib/python2.7/site-packages/mongoengine-0.8.7-py2.7.egg is being added to sys.path
  from pkg_resources import resource_stream

In [2]: rs = next(find_run_starts(scan_id=512))

In [3]: evd = next(find_event_descriptors(run_start_id=rs.id))

In [4]: %timeit find_events(descriptor_id=evd.id)
The slowest run took 26.67 times longer than the fastest. This could mean that an intermediate result is being cached 
1000 loops, best of 3: 288 µs per loop

In [5]: pwd
Out[5]: u'/home/edill'

In [6]: import os

In [7]: os.uname()
Out[7]: 
('Linux',
 'xf03idc-ws2',
 '3.2.0-4-amd64',
 '#1 SMP Debian 3.2.65-1+deb7u2',
 'x86_64')
```